House-of-glass/src/Contexts/Cart/CartContext.jsx

```diff
@@ -0,0 +1,169 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { cartReducer } from "./CartReducer";
+import Toast from "components/Toast/Toast";
+
+const CartContext = createContext();
+
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+  const initialCartState = { cart: [], cartCount: 0, cartSum: 0 };
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);
+
+  const token = localStorage.getItem("Token");
+  const header = {
+    headers: {
+      authorization: token,
+    },
+  };
+
+  const getCart = async () => {
+    try {
+      const response = await axios.get("/api/user/cart", header);
+      if (response.status === 200) {
+        cartDispatch({ type: "GET_CART", payload: response?.data?.cart });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    getCart();
+  }, []);
+
+  const addToCart = async (product) => {
```

Comment:
You can export all the functions related to API calls in a separate folder(naming to be decided by you) let's say API calls having these and then from an index.js file you can simply use it here.

Reply:
Yes, this is a good suggestion. Implementing it right away. Thanks Roshan.